docs/DesignDocs/enclave_openssl_support.md

```diff
@@ -0,0 +1,89 @@
+
+- Add OpenSSL as a git submodule to the OE SDK project.
+- Port the [build and code changes](https://github.com/openenclave/openenclave-openssl/blob/master/0001-Get-openssl-to-build-against-MUSL-headers.-Use-SGX-r.patch) necessary to build OpenSSL for use in an enclave to the OE SDK project.
+- Refactor the existing OE SDK enclave/crypto into enclave/crypto/mbedtls.
+- Port the prototype [oecrypto/src/crypto](https://github.com/openenclave/openenclave-openssl/tree/master/oecrypto/src/crypto) into OE SDK project enclave/crypto/openssl.
+  - The set of oecrypto methods used internally by the OE SDK that needs to be implemented in OpenSSL is expected to be at parity with enclave/crypto/mbedtls.
+- Port the prototype [tests](https://github.com/openenclave/openenclave-openssl/tree/master/test) for OpenSSL running in an enclave into OE SDK project as tests/openssl.
+- Integrate an OpenSSL variant into existing OE SDK tests that rely on crypto:
+  - [attestation_cert_apis](https://github.com/openenclave/openenclave-openssl/tree/master/oecrypto/test/attestation_cert_apis)
+  - attestation_plugin
+  - [crypto](https://github.com/openenclave/openenclave-openssl/tree/master/oecrypto/test/crypto)
+  - crypto_crls_cert_chains
+  - qeidentity
+  - report
+  - sealkey
+  - tls_e2e
+- Port the prototype [OpenSSL server in enclave sample](https://github.com/openenclave/openenclave-openssl/tree/master/sample/openssl_server) to OE SDK.
+  - This sample serves as the analog to the existing mbedTLS-specific file-encryptor sample.
+- Integrate an OpenSSL variant into existing OE SDK samples that rely on crypto:
+  - [attested_tls](https://github.com/openenclave/openenclave-openssl/tree/master/sample/attested_tls)
+  - data-sealing
+  - local_attestation
+  - remote_attestation
+- Document the support for OpenSSL in OE SDK, i.e. add documentation analogous to [MbedtlsSupport.md](https://github.com/openenclave/openenclave/blob/master/docs/MbedtlsSupport.md)
+
+Also, all prototypes have technical gaps that will need to be addressed for production purposes. A non-exhaustive list of such tasks would include:
+
+- [ ] Add ctr_drbg.c to the `oecryptoopenssl` library implementation for parity with `oecryptombedtls`.
+- [ ] Update OpenSSL code changes to use `oe_get_entropy()` for entropy pool seeding, and avoid modifying the OpenSSL drbg code.
+- [ ] Formalize additional syscall stubs needed for OpenSSL as needed for [ssl_stubs.c](https://github.com/openenclave/openenclave-openssl/blob/master/0001-Get-openssl-to-build-against-MUSL-headers.-Use-SGX-r.patch).
+- [ ] Update the OpenSSL target branch to the latest [OpenSSL_1_1_1-stable](https://github.com/openssl/openssl/tree/OpenSSL_1_1_1-stable) and resolve test coverage differences or breaking changes.
+- [ ] Evaluate the set of [broken OpenSSL tests](https://github.com/openenclave/openenclave-openssl/blob/master/test/tests.broken) when executed in an enclave.
```

Comment:
tests.broken was not complete. I think a bulk of work will be getting the openssl tests running inside enclaves.